commcare_connect/templates/opportunity/user_profile.html

```diff
@@ -84,20 +143,42 @@ <h6 class="card-title">Catchment Areas</h6>
       mapboxgl.accessToken = "{{ MAPBOX_TOKEN }}";
       const map = new mapboxgl.Map({
         container: 'user-visit-map',
-        style: 'mapbox://styles/mapbox/satellite-streets-v12',
+        style: 'mapbox://styles/mapbox/streets-v12', // Default to streets style
         center: [{{ lng_avg }}, {{ lat_avg }}],
         zoom: 14,
       });
+
       const userVisits = JSON.parse(document.getElementById('userVisits').textContent);
+      const userCatchments = JSON.parse(document.getElementById('userCatchments').textContent);
+
       userVisits.forEach(loc => {
         new mapboxgl.Marker()
           .setLngLat([loc.lng, loc.lat])
           .setPopup(new mapboxgl.Popup().setHTML(`${loc.entity_name}<br />${loc.visit_date}`))
           .addTo(map)
       })
-      addAccuracyCircles(map, userVisits);
-      const userCatchments = JSON.parse(document.getElementById('userCatchments').textContent);
-      addCatchmentAreas(map, userCatchments)
+
+      const setMapStyle = (styleId, targetElementId, otherElementId) => {
+        map.setStyle(styleId);
+        const targetElement = document.getElementById(targetElementId);
+        targetElement.classList.add('active', 'pe-none');
+
+        const otherElement = document.getElementById(otherElementId);
+        otherElement.classList.remove('active', 'pe-none');
+      };
+
+      document.getElementById('streets-v12').addEventListener('click', (e) => {
```

Comment:
You can use alpine.js `@click` binding on element directly. For e.g. @click="changeStyle('streets-v12')". Similarly for the other style. That will be more easier to understand/debug-latter.

```
    changeStyle(styleId) {
      const styles = {
        'streets-v12': 'mapbox://styles/mapbox/streets-v12',
        'satellite-streets-v12': 'mapbox://styles/mapbox/satellite-streets-v12'
      };
      
      if (this.currentStyle !== styleId) {
        this.map.setStyle(styles[styleId]);
        this.currentStyle = styleId;
      }
    },
```